java/java.project.ui/src/org/netbeans/spi/java/project/support/ui/CreateClassAndCopy.java

```diff
@@ -0,0 +1,172 @@
+
+        try {
+            Class c = Class.forName("org.openide.windows.TopComponent"); // NOI18N
+
+            // use reflection now
+            Method m = c.getMethod("getRegistry"); // NOI18N
+            Object o = m.invoke(null);
+
+            c = Class.forName("org.openide.windows.TopComponent$Registry"); // NOI18N
+
+            // use reflection now
+            m = c.getMethod("getActivated"); // NOI18N
+            o = m.invoke(o);
+
+            if (o instanceof ExplorerManager.Provider) {
+                return ((ExplorerManager.Provider) o).getExplorerManager();
+            }
+        } // exceptions from forName:
+        catch (ClassNotFoundException x) {
+        } catch (ExceptionInInitializerError x) {
+        } catch (LinkageError x) {
+        } // exceptions from getMethod:
+        catch (SecurityException x) {
+            t = x;
+        } catch (NoSuchMethodException x) {
+            t = x;
+        } // exceptions from invoke
+        catch (IllegalAccessException x) {
+            t = x;
+        } catch (IllegalArgumentException x) {
+            t = x;
+        } catch (InvocationTargetException x) {
+            t = x;
+        }
+
+        if (t != null) {
+            Logger.getLogger(PasteAction.class.getName()).log(Level.WARNING, null, t);
+        }
+
+        return null;
+    }
+
+    @Override
+    public void actionPerformed(ActionEvent ev) {
+        try {
+            Clipboard c = Toolkit.getDefaultToolkit().getSystemClipboard();
+            if(!c.isDataFlavorAvailable(DataFlavor.stringFlavor))return;
+            boolean containsClass = false;
+            String data = (String) c.getData(DataFlavor.stringFlavor);
+            String className = "";
+            String uncommented = data.replaceAll("(?:/\\*(?:[^*]|(?:\\*+[^*/]))*\\*+/)|(?://.*)","");// NOI18N
+            uncommented = uncommented.trim();
+            StringTokenizer st = new StringTokenizer(uncommented, " ");// NOI18N
+            if (st.hasMoreTokens() && st.nextToken().equals("public") && st.hasMoreTokens() && st.nextToken().equals("class") && st.hasMoreTokens()) {// NOI18N
+                String cname = st.nextToken().trim();
+                if (cname.charAt(cname.length() - 1) == '{') {
+                    cname = cname.substring(0, cname.length() - 1);
+                }
+                if (SourceVersion.isIdentifier(cname)) {
+                    containsClass = true;
+                    className = cname;
+                }
+            }
+            if (!containsClass) {
+                JOptionPane.showMessageDialog(null, "Code not valid to create class");
+                return;
+            }
+            Set<FileObject> files = this.context.files();
+            if (files.size() != 1) {
+                return;
+            }
+            String path = files.iterator().next().getPath();
+            String packageName = "";
+            File f = new File(path + "\\" + className + ".java");// NOI18N
```

Comment:
Using:
            File f = new File(path + "\\" + className + ".java");// NOI18N
will clearly not work on Linux, and probably not on Mac - the file separator for these is '/'. I would suggest to simply using FileObject all the way, unless there is a really compelling reason to not to.